File: ruby/fibonacci.rb
# Fibonacci Algorithm in Ruby
#
# Author: Elton Fonseca
# 

def fibonacci(term)
    fib0 = 1
    fib1 = 1

    puts fib0
    puts fib1

    for i in 3..term  do
        aux = fib0 + fib1
        fib0 = fib1
        fib1 = aux
        puts fib1
    end
end

fibonacci(10)

File: java/fibonacci/Fibonacci.java
public class Fibonacci
{
    public Fibonacci(int term)
    {
        int fib0 = 1, fib1 = 1, aux;

        System.out.print(fib0 + " " + fib1 + " ");
        for(int i = 3; i <= term; i++)
        {
            aux = fib0 + fib1;
            fib0 = fib1;
            fib1 = aux;

            System.out.print(fib1 + " ");
        }
        System.out.println();
    }
}